app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt

```diff
@@ -45,34 +48,50 @@ fun AuthScreen(
                 .imePadding()
                 .pointerInput(Unit) {
                     detectTapGestures {
-                        focusManager.clearFocus()
+                        focusManager.clearFocus() // Clear focus to close the keyboard
                     }
                 },
             verticalArrangement = Arrangement.Top,
             horizontalAlignment = Alignment.CenterHorizontally
         ) {
             FullScreenProgressBar(authViewModel.isProgressVisible)
-            LoginPageHeader()
-            when (currentScreen) {
-                AuthScreenState.Login -> LoginScreen(
-                    viewModel = authViewModel,
-                    onRegisterClicked = { currentScreen = AuthScreenState.Register },
-                    isLoginSucceed = { success, exception ->
-                        if (success && exception == null)
-                            navigator.navigate(MainScreens.Lectures)
-                    },
-                )
-
-                AuthScreenState.Register -> RegisterScreen(
-                    viewModel = authViewModel,
-                    onLoginClicked = { currentScreen = AuthScreenState.Login },
-                    isRegisterSucceed = { success, exception ->
-                        if (success && exception == null)
-                            navigator.navigate(MainScreens.Lectures)
-                    },
-                )
-            }
-
+            LoginPageHeader(authViewModel = authViewModel)
+            AnimatedContent(
+                targetState = currentScreen == AuthScreenState.Login,
```

Comment:
targetState is generic. I can put there anything I want but it will efect this:
content = { isVisible ->
                    if (isVisible) {
                        LoginScreen(
                            viewModel = authViewModel,
                            onRegisterClicked = { currentScreen = AuthScreenState.Register },
                            isLoginSucceed = { success ->
                                if (success)
                                    navigator.navigate(MainScreens.Lectures)
                            },
                        )
                    } else {
                        RegisterScreen(
                            viewModel = authViewModel,
                            onLoginClicked = { currentScreen = AuthScreenState.Login },
                            isRegisterSucceed = { success ->
                                if (success)
                                    navigator.navigate(MainScreens.Lectures)
                            },
                        )
                    }
                }

because targetState is boolean now then the lambda isVisible is boolean and that way I present the screen I want according to the animation